databricks/koalas/tests/test_dataframe.py

```diff
@@ -379,6 +380,67 @@ def test_rename_columns(self):
         self.assert_eq(kdf._internal.data_columns, ["('A', '0')", "('B', 1)"])
         self.assert_eq(kdf._internal.spark_df.columns, ["('A', '0')", "('B', 1)"])
 
+    def test_rename_dataframe(self):
+        kdf1 = ks.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
+        result_kdf = kdf1.rename(columns={"A": "a", "B": "b"})
+        self.assert_eq(result_kdf.columns, pd.Index(['a', 'b']))
+
+        result_kdf = kdf1.rename(index={1: 10, 2: 20})
+        self.assert_eq(result_kdf.index, pd.Index([0, 10, 20]))
+
+        def str_lower(s) -> str:
+            return str.lower(s)
+
+        result_kdf = kdf1.rename(str_lower, axis='columns')
+        self.assert_eq(result_kdf.columns, pd.Index(['a', 'b']))
+
+        def mul10(x) -> int:
+            return x * 10
+
+        result_kdf = kdf1.rename(mul10, axis='index')
+        self.assert_eq(result_kdf.index, pd.Index([0, 10, 20]))
+
+        result_kdf = kdf1.rename(columns=str_lower, index={1: 10, 2: 20})
+        self.assert_eq(result_kdf.columns, pd.Index(['a', 'b']))
+        self.assert_eq(result_kdf.index, pd.Index([0, 10, 20]))
+
+        idx = pd.MultiIndex.from_tuples([('X', 'A'), ('X', 'B'), ('Y', 'C'), ('Y', 'D')])
+        kdf2 = ks.DataFrame([[1, 2, 3, 4], [5, 6, 7, 8]], columns=idx)
+
+        result_kdf = kdf2.rename(columns=str_lower)
+        self.assert_eq(result_kdf.columns,
+                       pd.MultiIndex.from_tuples([('x', 'a'), ('x', 'b'), ('y', 'c'), ('y', 'd')]))
+
+        result_kdf = kdf2.rename(columns=str_lower, level=0)
+        self.assert_eq(result_kdf.columns,
+                       pd.MultiIndex.from_tuples([('x', 'A'), ('x', 'B'), ('y', 'C'), ('y', 'D')]))
+
+        result_kdf = kdf2.rename(columns=str_lower, level=1)
+        self.assert_eq(result_kdf.columns,
+                       pd.MultiIndex.from_tuples([('X', 'a'), ('X', 'b'), ('Y', 'c'), ('Y', 'd')]))
+
+        kdf3 = ks.DataFrame([[1, 2], [3, 4], [5, 6], [7, 8]], index=idx, columns=list('ab'))
+
+        # for spark 2.3, disable arrow optimization. Because koalas multi-index do not support
+        # arrow optimization in spark 2.3.
+
+        if LooseVersion(pyspark.__version__) <= LooseVersion("2.3"):
+            self.spark.conf.set('spark.sql.execution.arrow.enabled', False)
```

Comment:
Actually I asked to add it because toPandas's fallback seems added from Spark 2.4 as of https://github.com/apache/spark/commit/d6632d185e147fcbe6724545488ad80dce20277e

If we don't, it fails in Spark 2.3 as below:
```
                    return pd.DataFrame.from_records([], columns=self.columns)
            except Exception as e:
                msg = (
                    "Note: toPandas attempted Arrow optimization because "
                    "'spark.sql.execution.arrow.enabled' is set to true. Please set it to false "
                    "to disable this.")
>               raise RuntimeError("%s\n%s" % (_exception_message(e), msg))
E               RuntimeError: Unsupported type in conversion to Arrow: StructType(List(StructField(__index_level_0__,StringType,true),StructField(__index_level_1__,StringType,true)))
E               Note: toPandas attempted Arrow optimization because 'spark.sql.execution.arrow.enabled' is set to true. Please set it to false to disable this.
```